### PR TITLE
Mint | add mint support to new mint contract

### PR DESCRIPTION
**Pull request by FloppyDisck**
_Overview_
* New mint contract can now add assets
* Contract can now mint and burn accordingly
* Testers now sample this contract
* Replaced protocol msg traits with the ones included in secret_toolkit

_Details_
The micro_mint contract has been updated to a deploy ready state for a future replacement of the existing mint contract.

### CODE REVIEW

File: packages/shade_protocol/src/micro_mint/mod.rs [hunk 1]
```diff
@@ -48,6 +48,7 @@ pub enum HandleMsg {
     },
     RegisterAsset {
         contract: Contract,
+        burnable: bool,
```
Comment (DrPresident) [inline packages/shade_protocol/src/micro_mint/mod.rs:51]: When we can we should get this with the "token_config" and omit it from the parameters

File: packages/shade_protocol/src/snip20.rs [hunk 2]
```diff
@@ -8,4 +8,5 @@ use secret_toolkit::snip20::TokenInfo;
 pub struct Snip20Asset {
     pub contract: Contract,
     pub token_info: TokenInfo,
+    pub burnable: Option<bool>,
```
Comment (DrPresident) [inline packages/shade_protocol/src/snip20.rs:11]: Once we get this queried from the snip20, we should add the other values as well (that don't require extra queries) and make this not optional

**Review by DrPresident — APPROVED** [review]

File: packages/shade_protocol/src/micro_mint/mod.rs (hunk 1 above)
Comment (FloppyDisck) [reply packages/shade_protocol/src/micro_mint/mod.rs]: Agree, this change is now included in 6c71e

File: packages/shade_protocol/src/snip20.rs (hunk 2 above)
Comment (FloppyDisck) [reply packages/shade_protocol/src/snip20.rs]: Resolved in 6c71e

**Pull request merged**